### PR TITLE
fix: Handling of env from field.

### PR DESCRIPTION
**Pull request by iocanel**
Fixes: #473 

Also adds a test and eliminates duplicate env vars.

### CODE REVIEW

File: core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java [hunk 1]
```diff
@@ -41,68 +44,95 @@ public AddEnvVarDecorator(String deployment, String container, Env env) {
...
-   * If along with the envVars.getSecret() a envVars.getName()/envVars.getValue() has been specified they will be used to create an environment variable
-   * name after the envVars.getName(), that will point to the envVars.getSecret() entry named after the envVars.getValue().
-   * For example: (envVars.getName()=envVars.getName()1 envVars.getValue()=val1 and envVars.getSecret()=myenv.getSecret()). The added environment variable will be named envVars.getName()1 one
-   * and its envVars.getValue() will be read from envVars.getSecret() myenv.getSecret() by getting the envVars.getValue() of the entry named val1.
+   * Add an environment variable for the specified envVars.getSecret(). If along
+   * with the envVars.getSecret() a envVars.getName()/envVars.getValue() has been
+   * specified they will be used to create an environment variable name after the
+   * envVars.getName(), that will point to the envVars.getSecret() entry named
+   * after the envVars.getValue(). For example:
+   * (envVars.getName()=envVars.getName()1 envVars.getValue()=val1 and
```
Comment (geoand) [inline core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java]:
```suggestion
   * (envVars.getName()=envVars.getName() envVars.getValue()=val1 and
```

File: core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java [hunk 2]
```diff
@@ -41,68 +44,95 @@ public AddEnvVarDecorator(String deployment, String container, Env env) {
...
+   * envVars.getSecret()=myenv.getSecret()). The added environment variable will
+   * be named envVars.getName()1 one and its envVars.getValue() will be read from
```
Comment (geoand) [inline core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java]:
```suggestion
   * be named envVars.getName() one and its envVars.getValue() will be read from
```

File: core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java [hunk 3]
```diff
@@ -41,68 +44,95 @@ public AddEnvVarDecorator(String deployment, String container, Env env) {
...
-   * If along with the config map a envVars.getName()/envVars.getValue() has been specified they will be used to create an environment variable
-   * name after the envVars.getName(), that will point to the envVars.getSecret() entry named after the envVars.getValue().
-   * For example: (envVars.getName()=envVars.getName()1 envVars.getValue()=val1 and envVars.getConfigMap()=mymap). The added environment variable will be named envVars.getName()1 one
-   * and its envVars.getValue() will be read from configmap mymap by getting the envVars.getValue() of the entry named val1.
+   * Add an environment variable for the specified config map. If along with the
+   * config map a envVars.getName()/envVars.getValue() has been specified they
+   * will be used to create an environment variable name after the
+   * envVars.getName(), that will point to the envVars.getSecret() entry named
+   * after the envVars.getValue(). For example:
+   * (envVars.getName()=envVars.getName()1 envVars.getValue()=val1 and
```
Comment (geoand) [inline core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java]:
```suggestion
   * (envVars.getName()=envVars.getName() envVars.getValue()=val1 and
```

File: core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java [hunk 4]
```diff
@@ -41,68 +44,95 @@ public AddEnvVarDecorator(String deployment, String container, Env env) {
...
+   * envVars.getConfigMap()=mymap). The added environment variable will be named
+   * envVars.getName()1 one and its envVars.getValue() will be read from configmap
```
Comment (geoand) [inline core/src/main/java/io/dekorate/kubernetes/decorator/AddEnvVarDecorator.java]:
```suggestion
   * envVars.getName() one and its envVars.getValue() will be read from configmap
```

**Comment by iocanel** [comment]
@geoand: Completely removed the examples in the apidoc. Too much `find and replace` and I can't even understand it myself.

**Comment by geoand** [comment]
Actually, the test seems a little too minimal, no?

**Comment by iocanel** [comment]
> Actually, the test seems a little too minimal, no?

> Actually, the test seems a little too minimal, no?
I have absolutely no idea what you talking about. :-)
